models/users.js

```diff
@@ -178,26 +179,38 @@ const fetchPaginatedUsers = async (query) => {
   }
 };
 
-const fetchAllUsers = async () => {
+const fetchFilteredUsers = async (usernames = []) => {
   try {
     const dbQuery = userModel;
+    const filterdUsersWithDetails = [];
 
-    const snapshot = await dbQuery.get();
+    const groups = [];
+    for (let i = 0; i < usernames.length; i += BATCH_SIZE_IN_CLAUSE) {
+      groups.push(usernames.slice(i, i + BATCH_SIZE_IN_CLAUSE));
+    }
 
-    const allUsers = [];
+    // For each group, write a separate query
+    const promises = groups.map((group) => {
+      return dbQuery.where("github_id", "in", group).get();
+    });
 
-    snapshot.forEach((doc) => {
-      allUsers.push({
-        id: doc.id,
-        ...doc.data(),
-        phone: undefined,
-        email: undefined,
-        tokens: undefined,
-        chaincode: undefined,
+    const snapshots = await Promise.all(promises);
+
+    snapshots.forEach((snapshot) => {
+      snapshot.forEach((doc) => {
+        filterdUsersWithDetails.push({
+          id: doc.id,
+          ...doc.data(),
+          phone: undefined,
+          email: undefined,
+          tokens: undefined,
+          chaincode: undefined,
+        });
```

Comment:
In future, we should create a util or a DAO to ensure we can fetch user data with given explicit access clearance based on [data classifications](https://docs.aws.amazon.com/whitepapers/latest/data-classification/data-classification-models-and-schemes.html)